### DISCUSSION TITLE
Microcontroller stm32f401ccu6 overclocking

### DISCUSSION THREAD

**Discussion by brightproject** [stm32duino/Arduino_Core_STM32#2085] [Q&A] [answered]
Can be overclocked `stm32f401ccu6` up to 100 MHz?
I tried to play with numbers, as they say in the article
https://stm32f4-discovery.net/2014/11/overclock-stm32f4-device-up-to-250mhz/
- https://github.com/stm32duino/Arduino_Core_STM32/blob/987519a166179e3ada198364f79d388d05f75d3b/variants/STM32F4xx/F401CC(F-U-Y)_F401C(B-D-E)(U-Y)/variant_BLACKPILL_F401Cx.h#L143
- https://github.com/stm32duino/Arduino_Core_STM32/blob/987519a166179e3ada198364f79d388d05f75d3b/variants/STM32F4xx/F401CC(F-U-Y)_F401C(B-D-E)(U-Y)/variant_BLACKPILL_F401Cx.cpp#L97
The frequency was the same as 84 MHz and did not change, even sometimes decreased.

**Comment by fpistm** [comment]
Probably, never tested. If it does not change or decreased you probably made a mistake if you change directly in the variant cpp file.

**Reply by brightproject** [reply]
> Probably, never tested.

Is it generally possible to change the frequency using the `API`, directly from the program code?
I not found information in the [wiki](https://github.com/stm32duino/Arduino_Core_STM32/wiki/).

**Reply by fpistm** [reply, marked as answer]
https://github.com/stm32duino/Arduino_Core_STM32/wiki/Custom-definitions#systemclock_config
